panic for any command with new helm version

Seems the archlinux helm package got udpated and now when running helmfile it crash everytime

```sh
❯ helmfile lint                                                                                                                                                                                                      
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/roboll/helmfile/pkg/helmexec.getHelmVersion(0x15fab93, 0x4, 0xc000183150, 0x1b5f5e0, 0xc000028880, 0x447c01, 0xc0001d0101, 0xc00065dd40)
	/home/circleci/workspace/helmfile/pkg/helmexec/exec.go:68 +0x55c
github.com/roboll/helmfile/pkg/helmexec.New(...)
	/home/circleci/workspace/helmfile/pkg/helmexec/exec.go:106
github.com/roboll/helmfile/pkg/app.(*App).getHelm(0xc000476000, 0xc00024a000, 0x0, 0x0)
	/home/circleci/workspace/helmfile/pkg/app/app.go:555 +0x281
github.com/roboll/helmfile/pkg/app.(*App).visitStatesWithSelectorsAndRemoteSupport.func1(0xc00024a000, 0xc00024a000, 0x0, 0x0, 0xc0001bbb80)
	/home/circleci/workspace/helmfile/pkg/app/app.go:814 +0x41
github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1(0x16056e1, 0xd, 0xc0001aa090, 0x28, 0x0, 0x0)
	/home/circleci/workspace/helmfile/pkg/app/app.go:654 +0x7ec
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles.func1(0x0, 0x15f9a15)
	/home/circleci/workspace/helmfile/pkg/app/app.go:488 +0x9d
github.com/roboll/helmfile/pkg/app.(*App).within(0xc000476000, 0x15f9a15, 0x1, 0xc00065e478, 0xc00065e438, 0x2)
	/home/circleci/workspace/helmfile/pkg/app/app.go:430 +0x602
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles(0xc000476000, 0x0, 0x0, 0x25e6d30, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/circleci/workspace/helmfile/pkg/app/app.go:482 +0x2f2
github.com/roboll/helmfile/pkg/app.(*App).visitStates(0xc000476000, 0x0, 0x0, 0x25e6d30, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/circleci/workspace/helmfile/pkg/app/app.go:566 +0x10b
github.com/roboll/helmfile/pkg/app.(*App).visitStatesWithSelectorsAndRemoteSupport(0xc000476000, 0x0, 0x0, 0xc00065e798, 0xc00065e800, 0x1, 0x1, 0x1, 0xc0000367a0)
	/home/circleci/workspace/helmfile/pkg/app/app.go:821 +0x4c1
github.com/roboll/helmfile/pkg/app.(*App).ForEachState(0xc000476000, 0xc00065e808, 0xc00065e800, 0x1, 0x1, 0x15d89e0, 0xc0006e2a60)
	/home/circleci/workspace/helmfile/pkg/app/app.go:702 +0xb2
github.com/roboll/helmfile/pkg/app.(*App).Lint(0xc000476000, 0x1b88300, 0xc0006e2a60, 0xc0006e2a60, 0xc000183150)
	/home/circleci/workspace/helmfile/pkg/app/app.go:234 +0xa9
main.main.func6(0xc000476000, 0xc0001de9a0, 0x0, 0xc0006e2a50, 0x0)
	/home/circleci/workspace/helmfile/main.go:288 +0x6d
main.action.func1(0xc0001de9a0, 0x0, 0xc0001de9a0)
	/home/circleci/workspace/helmfile/main.go:698 +0xd4
github.com/urfave/cli.HandleAction(0x1374140, 0xc0006e2570, 0xc0001de9a0, 0xc00009a600, 0x0)
	/home/circleci/workspace/helmfile/vendor/github.com/urfave/cli/app.go:490 +0xc8
github.com/urfave/cli.Command.Run(0x15fac53, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x162c4e8, 0x27, 0x0, ...)
	/home/circleci/workspace/helmfile/vendor/github.com/urfave/cli/command.go:210 +0x9e8
github.com/urfave/cli.(*App).Run(0xc000448d00, 0xc00018c000, 0x2, 0x2, 0x0, 0x0)
	/home/circleci/workspace/helmfile/vendor/github.com/urfave/cli/app.go:255 +0x741
main.main()
	/home/circleci/workspace/helmfile/main.go:500 +0x354d
```

I had to fallback on helm `3.2.4` to make it work

--- a/pkg/helmexec/exec.go
+++ b/pkg/helmexec/exec.go
@@ -48,40 +48,28 @@ func NewLogger(writer io.Writer, logLevel string) *zap.SugaredLogger {
 	return zap.New(core).Sugar()
 }
 
-func getHelmVersion(helmBinary string, logger *zap.SugaredLogger, runner Runner) Version {
+// versionRegex matches versions like v1.1.1 and v1.1
+var versionRegex = regexp.MustCompile("v(?P<major>\\d+)\\.(?P<minor>\\d+)(?:\\.(?P<patch>\\d+))?")
 
-	// Autodetect from `helm verison`
-	bytes, err := runner.Execute(helmBinary, []string{"version", "--client", "--short"}, nil)
-	if err != nil {
-		panic(err)
+func parseHelmVersion(versionStr string) (Version, error) {
+	if len(versionStr) == 0 {
+		return Version{}, nil
 	}
 
-	if bytes == nil || len(bytes) == 0 {
-		return Version{}
+	matches := versionRegex.FindStringSubmatch(versionStr)
+	if len(matches) == 0 {
+		return Version{}, fmt.Errorf("error parsing helm verion '%s'", versionStr)
 	}
-
-	re := regexp.MustCompile("v(?P<major>\\d+)\\.(?P<minor>\\d+)\\.(?P<patch>\\d+)")
-	matches := re.FindStringSubmatch(string(bytes))
-
 	result := make(map[string]string)
-	for i, name := range re.SubexpNames() {
+	for i, name := range versionRegex.SubexpNames() {
 		result[name] = matches[i]
 	}
 
-	major, err := strconv.Atoi(result["major"])
-	if err != nil {
-		panic(err)
-	}
-
-	minor, err := strconv.Atoi(result["minor"])
-	if err != nil {
-		panic(err)
-	}
-
-	patch, err := strconv.Atoi(result["patch"])
-	if err != nil {
-		panic(err)
-	}
+	// We ignore errors because regex matches only integers
+	// If any of the parts does not exist - default "0" will be used
+	major, _ := strconv.Atoi(result["major"])
+	minor, _ := strconv.Atoi(result["minor"])
+	patch, _ := strconv.Atoi(result["patch"])
 
 	// Support explicit helm3 opt-in via environment variable
 	if os.Getenv("HELMFILE_HELM3") != "" && major < 3 {
@@ -89,21 +77,37 @@ func getHelmVersion(helmBinary string, logger *zap.SugaredLogger, runner Runner)
 			Major: 3,
 			Minor: 0,
 			Patch: 0,
-		}
+		}, nil
 	}
 
 	return Version{
 		Major: major,
 		Minor: minor,
 		Patch: patch,
+	}, nil
+}
+
+func getHelmVersion(helmBinary string, runner Runner) (Version, error) {
+
+	// Autodetect from `helm verison`
+	bytes, err := runner.Execute(helmBinary, []string{"version", "--client", "--short"}, nil)
+	if err != nil {
+		return Version{}, fmt.Errorf("error determining helm version: %w", err)
 	}
+
+	return parseHelmVersion(string(bytes))
 }
 
 // New for running helm commands
 func New(helmBinary string, logger *zap.SugaredLogger, kubeContext string, runner Runner) *execer {
+	// TODO: proper error handling
+	version, err := getHelmVersion(helmBinary, runner)
+	if err != nil {
+		panic(err)
+	}
 	return &execer{
 		helmBinary:       helmBinary,
-		version:          getHelmVersion(helmBinary, logger, runner),
+		version:          version,
 		logger:           logger,
 		kubeContext:      kubeContext,
 		runner:           runner,

--- a/pkg/helmexec/exec_test.go
+++ b/pkg/helmexec/exec_test.go
@@ -570,3 +570,50 @@ func Test_IsVersionAtLeast(t *testing.T) {
 	}
 
 }
+
+func Test_parseHelmVersion(t *testing.T) {
+	tests := []struct {
+		ver     string
+		want    Version
+		wantErr bool
+	}{
+		{
+			ver: "v1.2.3",
+			want: Version{
+				Major: 1,
+				Minor: 2,
+				Patch: 3,
+			},
+			wantErr: false,
+		},
+		{
+			ver: "v1.2",
+			want: Version{
+				Major: 1,
+				Minor: 2,
+				Patch: 0,
+			},
+			wantErr: false,
+		},
+		{
+			ver:     "v1",
+			wantErr: true,
+		},
+		{
+			ver:     "1.1.1",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.ver, func(t *testing.T) {
+			got, err := parseHelmVersion(tt.ver)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("parseHelmVersion() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("parseHelmVersion() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}

